src/wp_ecc_kmgmt.c

```diff
@@ -1004,8 +1004,17 @@ static int wp_ecc_validate(const wp_Ecc* ecc, int selection, int checkType)
        (void)checkType;
     #endif
         {
-            privDone = 1;
+            /* We may have a private key that does not correspond to the public
+             * key, which is fine as we are only being asked to check the
+             * public at this time. wc_ecc_check_key() always checks both if it
+             * has a private key, so we will fool it into only checking public
+             * key by manually setting the type */
+            origType = ecc->key.type;
+            if (ecc->key.type != ECC_PUBLICKEY) {
```

Comment:
Just assign anyway.
If threading is going to be an issue then copy out public key into new object. (I assume not but had to say it.)